### ISSUE TITLE
Auto fill from explorable rooms in spaces

### ISSUE THREAD

**Issue by Sleuth56** [vector-im/element-web#20349]
#### What would you like to do?
Use Element's auto fill feature to fill in room address that my account knows about but I'm a member of. 

#### Why would you like to do it?
Letting people know about rooms that I know of through element's auto fill feature is a great way of keeping people on topic in general rooms.

#### How would you like to achieve it?
Element would cache a local copy of all known rooms. These would be pulled from spaces that the user's account is apart of and sub spaces that they may or may not be apart of.

### Have you considered any alternatives?

Joining all rooms of a space will slow down sync pulls but would solve this issue. This is especially a problem for admins of large public spaces.
Manually typing room ids into chat isn't a good alternative.

### Additional context

This should be implemented across all platforms. Didn't want to make extra issues for each one. I can if that's helpful.

**Comment by HarHarLinks** [comment]
To clarify: you want autocomplete for room aliases (e.g. type `#`) to also offer rooms you "know about" but are not a member of?

**Comment by Sleuth56** [comment]
Exactly. I think the implementation should be agnostic to what data is given to it. That way in the future other rooms could be added to the auto complete function. The main data source right now would be rooms in spaces and sub spaces the user has joined.
